db/log_reader.h

```diff
@@ -123,6 +124,20 @@ class Reader {
     return !first_record_read_ && compression_type_record_read_;
   }
 
+  IOStatus Skip(uint64_t n) {
```

Comment:
According to the default posix file system's implementation, `Skip(n)` skips the next n bytes from the current file position: https://github.com/facebook/rocksdb/blob/389f75c86c084a477f5fb3730dd5bb39036dabac/env/io_posix.cc#L295

This is a bit from what this API does, so maybe use a different name that is more specific will be helpful, such as `SkipToOffsetAfterOpen(uint64_t offset)`, and enforce that this API is only called after Open, before and record is read and only called once. Something like asserting `firest_record_read_` before proceeding will be good.